tests/integration/cass_integration_test.go

```diff
@@ -0,0 +1,168 @@
+package integration
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/k8ssandra/k8ssandra/tests/integration/util"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	chartName  = "cass-operator"
+	chartPath  = "../../charts/k8ssandra/charts/cass-operator"
+	configPath = "C:\\Users\\Jeff Banks\\.kube\\config"
+
+	crdClusterYaml = "../../charts/k8ssandra-cluster/Chart.yaml"
+
+	defaultNamespace        = "default"
+	deleteCRDTimeoutSeconds = 15
+
+	k8ssandraNamespace  = "k8ssandra-ns"
+	operatorReleaseName = "k8ssandra-1"
+	clusterReleaseName  = "cluster-1"
+	clusterName         = "dc1"
+)
+
+// TestCassOperator performs basic installation of k8ssandra operator and c* cluster using setup & teardown.
+func TestCassOperatorAndClusterInstall(t *testing.T) {
+
+	kubeOptions := k8s.NewKubectlOptions("", "", k8ssandraNamespace)
+	helmOptions := &helm.Options{KubectlOptions: kubeOptions}
+
+	fmt.Println("helmOptions namespace is", helmOptions.KubectlOptions.Namespace)
+	require.NotNil(t, helmOptions)
+
+	// operator installation
+	installOperator(t, helmOptions, operatorReleaseName)
```

Comment:
If I follow the code correctly, it looks like the test does the following and in this order:

* Install k8ssandra chart
* Do some verification of k8ssandra release
* Delete objects installed as part of release
* Install k8ssandra-cluster chart
* Do some verification
* Delete objects installed as part of release

While we currently have the two charts, this is definitely a valid scenario. I don't think it is worth spending time on this scenario though since we are going to combine the charts.

I am not sure I see the need for all of the deletion code. We should be able to do cleanup simply by calling `helm uninstall`, albeit we need to uninstall in the correct order.

Reply:
@jsanda and @burmanm 

I've scaled-down the issue into a deconstructed example w/ an umbrella cleanup. 

This will help us to better brainstorm an overall approach.  

There's quite a bit going on in the this PR.  My bad.

**Basic Example**
```
kubeOptions := k8s.NewKubectlOptions("", "", "k8ssandra")
helmOptions := &helm.Options{KubectlOptions: kubeOptions}

// Cleanup
uninstallResult, uninstallErr := helm.RunHelmCommandAndGetOutputE(t, helmOptions, "uninstall", "-n", "k8ssandra", "k8ssandra", "k8ssandra/k8ssandra")
util.Log(t, "Helm", "Uninstall", uninstallResult, uninstallErr)

// Install the operator with all the defaults defined by the chart
installResult, installErr := helm.RunHelmCommandAndGetOutputE(t, helmOptions, "install", "-n", "k8ssandra",
	"k8ssandra", "k8ssandra/k8ssandra", "--create-namespace")
util.Log(t, "Helm", "Install", installResult, installErr)

// Basic verification of CRD for operator
lookupResult := util.LookupCRDByName(t, helmOptions, "cassandradatacenters.cassandra.datastax.com")
require.Equal(t, "customresourcedefinition.apiextensions.k8s.io/cassandradatacenters.cassandra.datastax.com",
	lookupResult)
```

**Context**
My understanding is that to perform verification of the installation (either operator or cluster), the previous associated resources (if existing) would need to be cleaned as part of the setup.  For example, if verifying that a CRD is being created properly, the older one would need to be removed. Otherwise, the validation check for its existence would always pass.  

The attempt was to cover the case where a developer has an existing `cassandradatacenters.cassandra.datastax.com` and a clean slate is required.  If that approach is out of scope due to combining of charts, that becomes an unneeded verification, and I agree, we could remove that type of cleanup and verification activities.  However, this may lead to some ugly state conditions for the developer if an operator's CRD exists and they can't remove it. 

The developer may try to wipe it out directly, which seems reasonable.
`kubectl delete crd/cassandradatacenters.cassandra.datastax.com`

In my experience, the response never returns, as a deadlock condition presents itself when finalizers are associated.
- [deadlock](https://github.com/kubernetes/kubernetes/issues/60538#issuecomment-369099998)

**Umbrella uninstall**
Below is a referenced issues I found related to installing without proper cleaning.  This led me down the path of the cleanup operations specific to k8ssandra `operator` and `cluster`.

- [resource exists](https://github.com/helm/helm/issues/7418#issuecomment-590840159)

**Specific Example**
> helm install -n k8ssandra k8ssandra k8ssandra/k8ssandra --create-namespace

```
Error: rendered manifests contain a resource that already exists. Unable to continue with install: ClusterRole "cass-operator" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; annotation 

validation error: key "meta.helm.sh/release-name" must equal "k8ssandra": current value is "k8ssandra-1"; annotation validation error: key "meta.helm.sh/release-namespace" must equal "k8ssandra": current value is "k8ssandra-ns"

```
**Question**
For our integration testing, are we wanting to allow customization in the test for any of the names (namespace and/or release name)?  

If not, then cleanup doesn't need to track for those differences by doing lookups of existing k8ssandra resources.  That would negate the need for value replacements used in the patching steps.  

Regardless, we may need to document how a developer can manually "clean slate" when the need arises.  Otherwise, the umbrella uninstall and install activities will fail due to lingering objects.

Reply:
>For our integration testing, are we wanting to allow customization in the test for any of the names (namespace and/or release name)?

I am not sure I understand. What do you mean by customization?

Reply:
> I am not sure I understand. What do you mean by customization?

Allowing the developer to customize their local release names or namespace names.

```
releaseName := "k8ssandra"
namespace := "k8ssandra"
```

Reply:
Yes, I would expect that the test should still pass if those values change.